Controller/Adminhtml/PayLink/Index.php

```diff
@@ -19,135 +19,103 @@
-
-        $services = [
-            'Name'             => 'payperemail',
-            'Action'           => 'PaymentInvitation',
-            'Version'          => 1,
-            'RequestParameter' => [
-                [
-                    '_'    => 'true',
-                    'Name' => 'MerchantSendsEmail',
-                ],
-                [
-                    '_'    => $order->getCustomerGender() ?? 1,
-                    'Name' => 'CustomerGender',
-                ],
-                [
-                    '_'    => $order->getCustomerEmail(),
-                    'Name' => 'CustomerEmail',
-                ],
-                [
-                    '_'    => $order->getCustomerFirstname(),
-                    'Name' => 'CustomerFirstName',
-                ],
-                [
-                    '_'    => $order->getCustomerLastname(),
-                    'Name' => 'CustomerLastName',
-                ],
-                [
-                    '_'    => $config->getPaymentMethod(),
-                    'Name' => 'PaymentMethodsAllowed',
-                ],
-            ]
-        ];
-
         $payment = $order->getPayment();
         $currentPayment = $payment->getMethod();
         $payment->setMethod('buckaroo_magento2_payperemail');
         $payment->save();
         $order->save();
 
-        $transactionBuilder = $this->transactionBuilderFactory->get('order');
-
-        $transactionBuilder->setOrder($order)
-            ->setServices($services)
-            ->setAdditionalParameter('fromPayLink', 1)
-            ->setAdditionalParameter('fromPayPerEmail', 1)
-            ->setMethod('TransactionRequest');
-
         try {
-            $transaction = $transactionBuilder->build();
+            $commandExecutor = $this->commandManagerPool->get('buckaroo');
 
-            $response = $this->gateway->authorize($transaction);
-            if (is_array($response[0]->Services->Service->ResponseParameter)) {
-                foreach ($response[0]->Services->Service->ResponseParameter as $parameter) {
-                    if ($parameter->Name == 'PayLink') {
-                        $payLink = $parameter->_;
-                    }
-                }
-            }
-        } catch (\Exception $e) {
-            $this->_messageManager->addErrorMessage($e->getMessage());
-        }
-
-        if (empty($payLink)) {
-            $this->_messageManager->addErrorMessage('Error creating PayLink');
-        } else {
-            $this->_messageManager->addSuccess(
-                __(
-                    'You PayLink <a href="%1">%1</a>',
-                    $payLink
-                )
+            $commandExecutor->executeByCode(
+                'paylink',
+                $payment,
+                [
+                    'amount' => $order->getGrandTotal()
+                ]
             );
+        } catch (\Exception $e) {
+            $this->messageManager->addErrorMessage($e->getMessage());
```

Comment:
better if we can filter the exceptions here, we may expose some critical information